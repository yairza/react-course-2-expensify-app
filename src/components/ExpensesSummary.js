import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => (
    <div>Viewing {expensesCount} expenses Totaling {numeral(expensesTotal / 100).format('$0,0.00')}
    </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
      expensesCount: visibleExpenses.length,
      expensesTotal : selectExpensesTotal(visibleExpenses)
    };
  };
  
  export default connect(mapStateToProps)(ExpensesSummary);