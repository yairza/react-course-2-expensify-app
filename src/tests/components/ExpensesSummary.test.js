import React from 'react';
import {shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses'; 

test('should render ExpensesSummary with expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses}  />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary without expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[]}  />);
    expect(wrapper).toMatchSnapshot();
});