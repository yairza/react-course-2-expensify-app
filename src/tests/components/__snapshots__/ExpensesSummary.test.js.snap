// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with expenses 1`] = `
<div>
  Viewing 
  3
   expenses Totaling 
  $1,141.95
</div>
`;

exports[`should render ExpensesSummary without expenses 1`] = `
<div>
  Viewing 
  0
   expenses Totaling 
  $0.00
</div>
`;
